pkg/utils/backoff.go

```diff
@@ -0,0 +1,58 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package utils
+
+import (
+	cRand "crypto/rand"
+	"math/big"
+	"math/rand"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/trace"
+)
+
+func init() {
+	seed, _ := cRand.Int(cRand.Reader, big.NewInt(1<<63-1))
+	rand.Seed(seed.Int64())
+}
+
+func Backoff(timeout time.Duration, f func() error) error {
+	defer trace.Trace("")()
+
+	maxTimer := time.NewTimer(timeout)
+	var err error
+
+	attempt := 0
+	sleepTime := time.Duration(rand.Int63n(int64(time.Second) / 4))
+
+	for ; ; sleepTime = sleepTime + time.Duration(rand.Int63n(int64(sleepTime+1))) {
```

Comment:
If my math is right, in worst case scenario (sleepTime randoms to 0 in line 31), this will generate ~20 attempts within first second, is that your intention?

Reply:
Although this scenario is very unlikely, you're right, and this is NOT my intention! I'll fix this by adding a minimum sleepTime of 250ms.

Reply:
Fixed!